test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/TagBuilderTest.cs

```diff
@@ -25,6 +25,26 @@ public class TagBuilderTest
         }
 
         [Theory]
+        [InlineData("attribute", "value", "<p attribute=\"HtmlEncode[[value]]\"></p>")]
+        [InlineData("attribute", null, "<p attribute=\"\"></p>")]
+        [InlineData("attribute", "", "<p attribute=\"\"></p>")]
+        public void AddAttributes_SpecialCases(string attributeKey, string attributeValue, string expectedOutput)
```

Comment:
Perhaps this would be `WriteTo_WriteEmptyAttribute_WhenValueIsNullOrEmpty` and move it Line 112 closer to the other `WriteTo` test?